src/org/omegat/core/team2/impl/GITRemoteRepository2.java

```diff
@@ -382,6 +382,32 @@ public static boolean deleteDirectory(File path) {
         return (path.delete());
     }
 
+    /**
+     * Retrieve default branch name from remote origin HEAD.
+     * @param repositoryUrl Remote repository URL
+     * @return default branch name, ordinary "main"(recent popular) or "master"(old default)
+     */
+    public static String getDefaultBranchName(final String repositoryUrl) {
+        try {
+            Map<String, Ref> gitMap = Git.lsRemoteRepository().setRemote(repositoryUrl).callAsMap();
+            Ref head = gitMap.get("HEAD");
+            if (head == null) {
+                return DEFAULT_LOCAL_BRANCH;
+            }
+            if (head.isSymbolic()) {
+                return head.getTarget().getName().substring(11);  // 11 = "refs/heads/".size
```

Comment:
Is the name guaranteed to start with `refs/heads/` in this case?

If we're not 100% sure, maybe we should check and throw a `RuntimeException` if it doesn't. Otherwise we will return garbage.

Reply:
I see you inlined `getBranchFromRef` into this method, but I liked it as a separate method because it's a single place to explain the magic number 11.

Reply:
from [git document](https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitrevisions.html)

> 
> <refname>, e.g. master, heads/master, refs/heads/master
> A symbolic ref name. E.g. master typically means the commit object referenced by refs/heads/master. If you happen to have both heads/master and tags/master, you can explicitly say heads/master to tell Git which one you mean. When ambiguous, a <refname> is disambiguated by taking the first match in the following rules:
> 
> 1. If $GIT_DIR/<refname> exists, that is what you mean (this is usually useful only for HEAD, FETCH_HEAD, ORIG_HEAD, MERGE_HEAD and CHERRY_PICK_HEAD);
> 2. otherwise, refs/<refname> if it exists;
> 3. otherwise, refs/tags/<refname> if it exists;
> 4. otherwise, refs/heads/<refname> if it exists;
> 5. otherwise, refs/remotes/<refname> if it exists;
> 6. otherwise, refs/remotes/<refname>/HEAD if it exists.
> 
> HEAD names the commit on which you based the changes in the working tree. FETCH_HEAD records the branch which you fetched from a remote repository with your last git fetch invocation. ORIG_HEAD is created by commands that move your HEAD in a drastic way, to record the position of the HEAD before their operation, so that you can easily change the tip of the branch back to the state before you ran them. MERGE_HEAD records the commit(s) which you are merging into your branch when you run git merge. CHERRY_PICK_HEAD records the commit which you are cherry-picking when you run git cherry-pick.
> 
> Note that any of the refs/* cases above may come either from the $GIT_DIR/refs directory or from the $GIT_DIR/packed-refs file. While the ref name encoding is unspecified, UTF-8 is preferred as some output processing may assume ref names in UTF-8.

Things is not easy.
>

Reply:
Here is another approach. It also works.

```
    /**
     * Retrieve default branch name from repository.
     * @param repository target repository.
     * @return default branch name, ordinary "main"(recent popular) or "master"(old default)
     */
    public static String getDefaultBranchName(final Repository repository) {
        try {
            return getBranchFromRef(repository.exactRef(Constants.HEAD).getTarget().getName());
        } catch (IOException e) {
            Log.log(e);
        }
        return DEFAULT_LOCAL_BRANCH;
    }
```

Reply:
I've replace `getBranchFromRef` with `shortenRefName()` of JGit library.